### DISCUSSION TITLE
Exclude a directory in the top level by name, but don't exclude others with the same name?

### DISCUSSION THREAD

**Discussion by vroad** [oxsecurity/megalinter#3242] [Q&A]
For example, how can I exclude a directory named "foo" in the top level directory, but include the same named directories found elsewhere?

EXCLUDED_DIRECTORIES are excluded at any nested level, not just the directory found in the top level directory.
I'm not sure FILTER_REGEX_EXCLUDE support regex special characters like `^`.

MegaLinter config like this didn't work for me.

```yaml
FILTER_REGEX_EXCLUDE: ^foo/
# FILTER_REGEX_EXCLUDE: (?m)^foo/
```

```yaml
FILTER_REGEX_EXCLUDE: foo
```

This works but excludes `foo` at any nested level.

**Comment by nvuillam** [comment]
Please can you try `(^foo)` ? :)
it is possible that to keep ascending compatibility with v6, who used files with absolute path, something might be wrong

![image](https://github.com/oxsecurity/megalinter/assets/17500430/a1f6d991-1e21-44b1-9bb5-5bd3019bd19a)

If that's the case, we could add a REGEX_STRICT config that would disable the "or"

**Comment by vroad** [comment]
`(^foo)` didn't work either. Since this regex lacks trailing `/`, doesn't this also exclude other directories whose name starting with foo, like `foo1/some_file`, `foo2/another_file`, etc?

```yml
FILTER_REGEX_EXCLUDE: (^foo)
```
